src/utils/auth-channel.js

```diff
@@ -39,17 +39,56 @@ export default class AuthChannel {
       channel
         .join()
         .receive("ok", () => {
-          channel.on("auth_credentials", async ({ credentials: token, payload: payload }) => {
-            await this.handleAuthCredentials(payload.email, token);
-            resolve();
-          });
-
-          channel.push("auth_verified", { token: authToken, payload: authPayload });
+          if (origin === "oidc") {
+            channel
+              .push("auth_verified", { token: authToken, payload: authPayload })
+              .receive("ok", resolve)
+              .receive("error", reject);
+          } else {
+            channel.push("auth_verified", { token: authToken, payload: authPayload });
+            channel.on("auth_credentials", async ({ credentials: token, payload: payload }) => {
+              await this.handleAuthCredentials({ email: payload.email }, token);
+              resolve();
+            });
```

Comment:
Probably best to keep the previous order and register the handler before pushing the message.